mutiny-core/src/lib.rs

```diff
@@ -2117,47 +2119,39 @@ impl<S: MutinyStorage> MutinyWallet<S> {
-                            }),
-                            NwcProfileTag::Subscription,
-                            vec![Method::PayInvoice], // subscription only needs pay invoice
-                        )
-                        .await?
-                        .nwc_uri
-                } else {
-                    self.nostr
-                        .create_new_nwc_profile(
-                            ProfileType::Reserved(ReservedProfile::MutinySubscription),
-                            SpendingConditions::RequireApproval,
-                            NwcProfileTag::Subscription,
-                            vec![Method::PayInvoice], // subscription only needs pay invoice
-                        )
-                        .await?
-                        .nwc_uri
-                };
+        if profile_opt.is_none() {
+            log_debug!(self.logger, "Did not find a mutiny+ nwc profile");
+            // profile with the reserved index does not exist, create a new one
+            let nwc = if autopay {
+                self.nostr
+                    .create_new_nwc_profile(
+                        ProfileType::Reserved(ReservedProfile::MutinySubscription),
+                        SpendingConditions::Budget(BudgetedSpendingConditions {
+                            budget: 21_000,
+                            single_max: None,
+                            payments: vec![],
+                            period: BudgetPeriod::Month,
+                        }),
+                        NwcProfileTag::Subscription,
+                        vec![Method::PayInvoice], // subscription only needs pay invoice
+                    )
+                    .await?
+                    .nwc_uri
+            } else {
+                self.nostr
+                    .create_new_nwc_profile(
+                        ProfileType::Reserved(ReservedProfile::MutinySubscription),
+                        SpendingConditions::RequireApproval,
+                        NwcProfileTag::Subscription,
+                        vec![Method::PayInvoice], // subscription only needs pay invoice
+                    )
+                    .await?
+                    .nwc_uri
+            };
 
-                if let Some(nwc) = nwc {
-                    // only should have to submit the NWC if never created locally before
-                    subscription_client.submit_nwc(nwc).await?;
-                }
-            }
-            Some(profile) => {
-                if profile.tag != NwcProfileTag::Subscription {
-                    let mut nwc = profile.clone();
-                    nwc.tag = NwcProfileTag::Subscription;
-                    self.nostr.edit_nwc_profile(nwc)?;
-                }
+            if let Some(nwc) = nwc {
+                // only should have to submit the NWC if never created locally before
+                subscription_client.submit_nwc(nwc).await?;
             }
         }
```

Comment:
why delete this?

Reply:
Because it's not needed. Why keep it?

Reply:
We use this in the front end to mark it as a subscription so we know to not display the nwc connection string

Reply:
It should already be marked correctly and that's not how the frontend handles it. https://github.com/MutinyWallet/mutiny-web/blob/a4981cb8cf0d8afe6f7e5d540fac0cc6b9c858d4/src/routes/settings/Connections.tsx#L110-L118

Reply:
ah, its for the open in client buttons

https://github.com/MutinyWallet/mutiny-web/blob/a4981cb8cf0d8afe6f7e5d540fac0cc6b9c858d4/src/routes/settings/Connections.tsx#L157

Reply:
> It should already be marked correctly

ah okay, this was for fixing old profiles, that's fine then